app/src/main/java/com/duckduckgo/app/brokensite/BrokenSiteViewModel.kt

```diff
@@ -103,7 +105,13 @@ class BrokenSiteViewModel(
 
     fun onSubmitPressed(webViewVersion: String) {
         if (url.isNotEmpty()) {
-            val brokenSite = getBrokenSite(webViewVersion)
+
+            val brokenSite = if (trackingLinkDetector.lastTrackingLinkInfo?.destinationUrl == url) {
+                getBrokenSite(trackingLinkDetector.lastTrackingLinkInfo!!.trackingLink, webViewVersion)
```

Comment:
one way to avoid using `!!` is to use a temp `val`

`val lastTrackinInfo = trackingLinkDetector.lastTrackingLinkInfo`

and then use that one, it's not mutable, so you won't need to use `!!`.